pkg/apis/extensions/v1beta1/types.go

```diff
@@ -253,8 +253,13 @@ type DeploymentSpec struct {
 	// Indicates that the deployment is paused and will not be processed by the
 	// deployment controller.
 	Paused bool `json:"paused,omitempty" protobuf:"varint,7,opt,name=paused"`
+
 	// The config this deployment is rolling back to. Will be cleared after rollback is done.
 	RollbackTo *RollbackConfig `json:"rollbackTo,omitempty" protobuf:"bytes,8,opt,name=rollbackTo"`
+
+	// The maximum time in seconds for a deployment to make progress before it
+	// is considered to be failed. This is not set by default.
```

Comment:
Explain what it means to be failed in an extra sentence or two.  This will probably the first point people go to to find out what a failed deployment *is*.